commit 6d601cf09330ab79ab54d0db9325885c8c4c8dbf
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-05-02T10:55:54-07:00 (5 months ago)
    
    TRITON-1642 migration tests are failing with "No compute resources available" error
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/workflows/vm-migration/begin.js b/lib/workflows/vm-migration/begin.js
index 693a250..de6b039 100644
--- a/lib/workflows/vm-migration/begin.js
+++ b/lib/workflows/vm-migration/begin.js
@@ -336,11 +336,15 @@ function allocateServer(job, cb) {
     });
 
 
-    // Make sure our VM is placed on a CN away from the old VM.
-    job.vmPayload.locality = {
-        far: job.params.vm.uuid,
-        strict: true
-    };
+    // When not overiding the vm uuid (i.e. for testing purpuses, to provision
+    // to the the same server) make sure the target VM will be placed on a CN
+    // away from the source VM.
+    if (record.vm_uuid === record.target_vm_uuid) {
+        job.vmPayload.locality = {
+            far: job.params.vm.uuid,
+            strict: true
+        };
+    }
 
     // The vmPayload must use vm.vm_uuid (not sure why it's different), anyway,
     // copy it across.
diff --git a/package.json b/package.json
index f74d152..c1c484d 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.8.4",
+  "version": "9.8.5",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/test/lib/migration.js b/test/lib/migration.js
index 9e59947..d19a317 100644
--- a/test/lib/migration.js
+++ b/test/lib/migration.js
@@ -141,6 +141,8 @@ function TestMigrationCfg(test, cfg) {
                     return s.status === 'running' &&
                         s.sysinfo && s.sysinfo['System Type'] !== 'Virtual';
                 });
+                t.ok(availableServers.length > 0,
+                    'number of available servers: ' + availableServers.length);
                 if (availableServers.length < 2) {
                     migrationUuidOverride = true;
                 }
